datasets/ba/companies/crawler.py

```diff
@@ -0,0 +1,522 @@
+        "x01": "-1",
+        "x02": "-1",
+        "x03": "-1",
+    }
+
+    RETRIEVE_PAYLOAD = {
+        "p": f"186:3:{secret_param}:FLOW_PPR_OUTPUT_R16339113485096783_pg_"
+        + "R_16339113485096783:NO",
+        "pg_max_rows": "5000",
+        "pg_min_row": "1",
+        "pg_rows_fetched": "undefined",
+    }
+
+    context.fetch_text(
+        url=TOUCH_URL,
+        method="POST",
+        data=TOUCH_PAYLOAD1,
+        headers=HEADER,
+        cache_days=CACHE_DAYS,
+    )
+
+    context.fetch_text(
+        url=TOUCH_URL,
+        method="POST",
+        data=TOUCH_PAYLOAD2,
+        headers=HEADER,
+        cache_days=CACHE_DAYS,
+    )
+
+    result = context.fetch_html(
+        url=RETRIEVE_URL,
+        method="POST",
+        data=RETRIEVE_PAYLOAD,
+        headers=HEADER,
+        cache_days=CACHE_DAYS,
+    )
+
+    rows = result.findall(".//tr")
+
+    records = []
+    for row in rows:
+        if row.find(".//td/a") is None:
+            continue
+        record = {}
+        record["city"] = city.get("city", "")
+        record["registration_number"] = row[0].text
+        record["name"] = row[1][0].text
+        record["abbreviation"] = row[2][0].text
+        record["address"] = row[3].text
+        record["date_of_last_decision"] = row[4].text
+        record["details_url"] = urljoin(BASE_URL, row[1][0].attrib["href"])
+        record["key"] = "::".join(
+            [record.get("registration_number", {}), record.get("name", {})]
+        )
+
+        records.append(record)
+
+    return records
+
+
+def crawl_details(context: Context, record: Dict[str, str]) -> None:
+    """
+    Fetches the details of a company from the website.
+    Args:
+        context: The context object for the current dataset.
+        record: The record to fetch the details for.
+    Returns:
+        The details of the company as a dict.
+    """
+    details_page = context.fetch_html(record["details_url"], headers=HEADER)
+
+    legal_form = details_page.xpath(
+        "//td[contains(text(),"
+        + ' "Legal form of organization")'
+        + "]/following-sibling::td/text()"
+    )
+    if legal_form:
+        record["legal_form"] = legal_form[0]
+
+    address_add = details_page.xpath(
+        "//td[contains(text()," + ' "Address")' + "]/following-sibling::td/text()"
+    )
+    if address_add:
+        record["address_additional"] = address_add[0]
+
+    status = details_page.xpath(
+        "//td[contains(text(),"
+        + ' "Status (Bankruptcy – YES/NO)")'
+        + "]/following-sibling::td/text()"
+    )
+    if status:
+        record["status_bankruptcy"] = status[0]
+
+    uin = details_page.xpath(
+        "//td[contains(text(),"
+        + ' "Unique Identification Number")'
+        + "]/following-sibling::td/text()"
+    )
+    # JD: Jedinstveni identifikacioni broj - JIB or UIN
+    # https://www.vatify.eu/bosnia-and-herzegovina-vat-number.html
+    if uin:
+        record["unique_id"] = uin[0]
```

Comment:
what's the question? or does this speak to the one where say yeah let's also add this as a second regno?

Reply:
The question is in the another comment :)